src/app/georeferencer/qgsgeorefmainwindow.cpp

```diff
@@ -1291,8 +1291,14 @@ bool QgsGeoreferencerMainWindow::loadGCPs( /*bool verbose*/ )
 
     QgsPointXY mapCoords( ls.at( 0 ).toDouble(), ls.at( 1 ).toDouble() ); // map x,y
     QgsPointXY pixelCoords( ls.at( 2 ).toDouble(), ls.at( 3 ).toDouble() ); // pixel x,y
-    QgsCoordinateReferenceSystem proj( ls.at( 8 ) );
-    if ( ls.count() == 5 || ls.count() == 9 )
+
+    QgsCoordinateReferenceSystem proj;
+    if ( ls.count() == 9 )
+      proj = QgsCoordinateReferenceSystem( ls.at( 8 ) );
+    else
+      proj = QgsProject::instance()->crs();
+
```

Comment:
`ls.count()` will hardly ever be 9 with a valid crs wkt definition, since it contains a few dozens of commas in it!

Would it be possible to at least store only the epsg code instead?

Reply:
That would only work in some circumstances -- it's possible to have an empty authid value for eg custom projections

Reply:
^ as Nyall stated this is why wkt is used, but you're right, I'm not sure how it's parsed I'll need to debut to validate. otherwise changing the splitting might be needed.

Reply:
This sounds like a pretty exotic use case scenario though...

Reply:
It's not that uncommon actually -- especially since the the post proj 6 world we are stricter about matching a crs definition to a known crs. And then you have other things like compoundcrs or boundcrs types which also won't return anything for authid... Suffice to say, we can't ignore this situation. 

That said, a cleaner solution would be good! I wonder if we could avoid exporting the crs for all lines if it's identical and instead putting it in a single "metadata" style line at the start of the file (eg prefixed with #)

Reply:
@nyalldawson do you propose to transform all points in a single crs before exporting? This would ensure that a single projection is needed to repopulate the table. Though this may be confusing for manual edits.

Reply:
No I mean when we write the gcp file we check if all the point crses are identical, and if so we just put a single crs metadata line into the file...

Reply:
The cleanest approach would be to either store points on a single crs (as per my suggestion in https://github.com/qgis/QGIS/pull/39584#issuecomment-755705406) or be future proof and upgrade the file format to an xml or something (with a different extension for backwards compatibility)

Reply:
that or I could replace the commas in the string on save and load to prevent bad splits, though xml might be nicer.

Reply:
String replacement is a hack, not a solid solution.
If we do change the format at some point, we should consider using a format that can be loaded-modified as a qgis layer.